### DISCUSSION TITLE
CloudFront "Public" / "Don't use OAI" option in S3 origin

### DISCUSSION THREAD

**Discussion by JackPriceBurns** [aws/aws-cdk#23539] [Q&A]
I'm using the Go CDK and when I omit the OriginAccessIdentity when creating an S3 origin on a CloudFront distribution, it seems to create one anyway instead of not setting it which is what I actually wanted.

[An issue was raised about this](https://github.com/aws/aws-cdk/issues/16200) however it was closed.

When I run this
```
	jacksBucket := awss3.Bucket_FromBucketName(stack, jsii.String("JacksBucket"), jsii.String(bucketName))

	// Create new origin
	jacksOrigin := awscloudfrontorigins.NewS3Origin(jacksBucket, &awscloudfrontorigins.S3OriginProps{
		// Get content from the / directory in the bucket.
		OriginPath: jsii.String("/"),
		// OriginAccessIdentity: would normally be set here, I've omitted it
	})

	// Create new behaviour
	cf.AddBehavior(jsii.String("/bucket*"), jacksOrigin, &awscloudfront.AddBehaviorOptions{
		AllowedMethods:       awscloudfront.AllowedMethods_ALLOW_GET_HEAD_OPTIONS(),
		CachedMethods:        awscloudfront.CachedMethods_CACHE_GET_HEAD(),
		CachePolicy:          awscloudfront.CachePolicy_CACHING_DISABLED(),
		Compress:             jsii.Bool(true),
		OriginRequestPolicy:  awscloudfront.OriginRequestPolicy_ALL_VIEWER(),
		ViewerProtocolPolicy: awscloudfront.ViewerProtocolPolicy_REDIRECT_TO_HTTPS,
	})
```

Since I have not specified an OriginAccessIdentity in the S3Origin I would expect to see this 
<img width="822" alt="Screenshot 2023-01-03 at 1 18 23 pm" src="https://user-images.githubusercontent.com/5484580/210364857-d6602886-54d6-409b-96b6-87f981c93f03.png">

However, I see this
<img width="832" alt="Screenshot 2023-01-03 at 1 15 29 pm" src="https://user-images.githubusercontent.com/5484580/210364899-edfe624e-c59d-4062-8310-134e44807328.png">

The AWS Management Console makes it easy to just select "Public" which removes the OAI and [in CloudFormation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cloudfront-distribution-s3originconfig.html#cfn-cloudfront-distribution-s3originconfig-originaccessidentity:~:text=If%20you%20want%20viewers%20to%20be%20able%20to%20access%20objects%20using%20either%20the%20CloudFront%20URL%20or%20the%20Amazon%20S3%20URL%2C%20specify%20an%20empty%20OriginAccessIdentity%20element.) it also states that leaving this empty means that it doesn't use an OAI, so this feels like an issue with the CDK.

The reason I put this as a discussion is that I'm unsure if this is a bug or if I'm using the CDK wrong? Should I be explicitly setting the origin access to Public some other way? I've looked through all the properties for creating an S3 origin and access identity but can't see anything obvious that would indicate this.

**Comment by JackPriceBurns** [comment]
I've found a way around this issue by using cfn override... However, this is not ideal. This references the origin by an index which could change if anyone adds more origins. I can't seem to find any other way to solve this, I'm open to any suggestions.

```
	// Delete the S3OriginConfig as CDK generates it when we don't want it.
	cfnDistribution := cf.Node().DefaultChild().(awscloudfront.CfnDistribution)
	cfnDistribution.AddPropertyOverride(jsii.String("DistributionConfig.Origins.1.S3OriginConfig.OriginAccessIdentity"), "")
```

I think this warrants an issue being created as this is functionality provided by both the management console and CloudFormation that isn't possible without a hacky override.

**Comment by peterwoodworth** [comment]
Thanks for creating this discussion @JackPriceBurns, there doesn't seem to be a clear way to do this without overrides.

I've reopened the original issue, let's keep the discussion there 🙂
